File: Mvc/Models/Disk.cs
using System.ComponentModel.DataAnnotations;

namespace Mvc.Models
{
    public class Disk
    {
        public int Id { get; set; }
        [Display(Name = "Название")]
        public string Name { get; set; }
        [Display(Name = "Размер")]
        public int SizeGb { get; set; }
        [Display(Name = "Скорость записи")]
        public int WriteSpeedMbps { get; set; }
        [Display(Name = "Скорость чтения")]
        public int ReadSpeedMbps { get; set; }
        [Display(Name = "Цена")]
        public decimal Price { get; set; }
        [Display(Name = "Тип")]
        public DiskType Type { get; set; }
        [Display(Name = "Формат")]
        public DiskForm Form { get; set; }
    }

    public enum DiskType
    {
        SSD,
        HDD,
    }

    public enum DiskForm
    {
        [Display(Name = "SATA 2.5\"")]
        Sata25,
        [Display(Name = "SATA 3.5\"")]
        Sata35,
        [Display(Name = "M.2")]
        M2,
    }
}

File: react-spa/src/types/Disk.ts
interface Disk {
    id: number;
    name: string;
    sizeGb: number;
    writeSpeedMbps: number;
    readSpeedMbps: number;
    price: number;
    type: DiskType;
    form: DiskForm;
}

enum DiskType {
    SSD = 0,
    HDD = 1,
}

const DISK_TYPE_DISPLAY = {
    [DiskType.SSD]: 'SSD',
    [DiskType.HDD]: 'HDD',
};

enum DiskForm {
    Sata25 = 0,
    Sata35 = 1,
    M2 = 2,
}

const DISK_FORM_DISPLAY = {
    [DiskForm.Sata25]: 'SATA 2.5"',
    [DiskForm.Sata35]: 'SATA 3.5"',
    [DiskForm.M2]: 'M.2',
};

export type { Disk };
export { DiskType, DISK_TYPE_DISPLAY, DiskForm, DISK_FORM_DISPLAY };
